winpr/libwinpr/library/library.c

```diff
@@ -209,9 +232,45 @@ HMODULE GetModuleHandleW(LPCWSTR lpModuleName)
 
 DWORD GetModuleFileNameW(HMODULE hModule, LPWSTR lpFilename, DWORD nSize)
 {
-	WLog_ERR(TAG, "%s is not implemented", __FUNCTION__);
-	SetLastError(ERROR_CALL_NOT_IMPLEMENTED);
-	return 0;
+	WCHAR* wname = NULL;
+	char* name = NULL;
+	int csize;
+	DWORD status;
+	{
+		csize = ConvertFromUnicode(CP_UTF8, 0, lpFilename, -1, &name, 0, NULL, NULL);
```

Comment:
This looks wrong ... why do we convert an uninitialized input buffer?

Reply:
`lpFilename` is a function argument, `name` will be allocated by this call. what do you mean?

Reply:
```lpFilename``` is an output buffer - I don't understand why we convert that buffer before it is even initialized